### PR TITLE
app-catalog: Update to headlamp-plugin@0.8.0-alpha.x and mui5

### PR DESCRIPTION
**Pull request by illume**
Updates app-catalog so it works with the main branch.

Note: You have to use this headlamp branch until it is merged in: `mui-fixes` https://github.com/headlamp-k8s/headlamp/pull/1595

## How to test

Install this plugin with `npm start`.

Then run the headlamp app.
```
cd app
npm i
npm start
```

- [ ] Please test the app-catalog plugin manually with latest headlamp main branch
- [ ] check the AutoComplete works

### CODE REVIEW

**Comment by yolossn** [comment]
image
The app list page is failing with the above logs.

**Comment by illume** [comment]
@yolossn Sorry. Same with the other plugin... You have to use this headlamp branch until it is merged in: `mui-fixes` https://github.com/headlamp-k8s/headlamp/pull/1595

**Comment by ashu8912** [comment]
Screenshot 2023-12-20 at 5 19 05 PM
Go to app-catalog list view click on any of the chart install button and the app will crash

**Comment by illume** [comment]
Thanks @ashu8912 I'm looking into it.

**Comment by yolossn** [comment]
@illume I ran into the same issue, but I am not sure if it is happening due to the changes that are introduced in this PR.

**Comment by illume** [comment]
Thanks @ashu8912 and @yolossn

File: app-catalog/src/components/releases/ReleaseList.stories.tsx [hunk 1]
```diff
@@ -53,6 +56,19 @@ const initialState = {
   },
 };
 
+// const mockStore = configureStore({
```
Comment (illume) [inline app-catalog/src/components/releases/ReleaseList.stories.tsx]: Some commented out code needs removing

File: app-catalog/src/components/releases/ReleaseList.stories.tsx [hunk 2]
```diff
@@ -1,18 +1,21 @@
+// ignore all the eslint issues
+/* eslint-disable */
```
Comment (illume) [inline app-catalog/src/components/releases/ReleaseList.stories.tsx]: this was left in for debugging... will remove

**Review by ashu8912 — APPROVED** [review]

**Comment by illume** [comment]
I removed some commented out debugging code.

**Pull request merged**